Lib/ufo2ft/instructionCompiler.py

```diff
@@ -0,0 +1,310 @@
+
+        # Keep track of which component has the USE_MY_METRICS flag
+        # and whether any component lib contains the useMyMetrics key
+        use_my_metrics_comp = None
+        lib_contains_use_my_metrics_key = False
+
+        for i, c in enumerate(ttglyph.components):
+            # Set OVERLAP_COMPOUND on the first component only
+            if i == 0 and TRUETYPE_OVERLAP_KEY in glyph.lib:
+                if glyph.lib.get(TRUETYPE_OVERLAP_KEY, False):
+                    c.flags |= OVERLAP_COMPOUND
+                else:
+                    c.flags &= ~OVERLAP_COMPOUND
+
+            # Check if we have information about the current component in the glyph lib
+            ufo_component_id = glyph.components[i].identifier
+            if ufo_component_id is None:
+                # No information about component flags is stored in the UFO.
+                # We don’t modify the flags. Two flags have already been set elsewhere:
+                # - ROUND_XY_TO_GRID is set in TTGlyphPointPen.glyph() called from
+                #                    OutlineTTFCompiler.compileGlyphs()
+                # - USE_MY_METRICS   is set in OutlineTTFCompiler.setupTable_glyf()
+                continue
+
+            if (
+                OBJECT_LIBS_KEY in glyph.lib
+                and ufo_component_id in glyph.lib[OBJECT_LIBS_KEY]
+                and (
+                    TRUETYPE_ROUND_KEY in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                    or TRUETYPE_METRICS_KEY
+                    in glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+                )
+            ):
+                component_lib = glyph.lib[OBJECT_LIBS_KEY][ufo_component_id]
+
+                # ROUND_XY_TO_GRID
+
+                # https://github.com/googlefonts/ufo2ft/pull/425 recommends
+                # to always set the ROUND_XY_TO_GRID flag, so we only
+                # unset it if explicitly done so in the lib
+                if not component_lib.get(TRUETYPE_ROUND_KEY, True):
+                    c.flags &= ~ROUND_XY_TO_GRID
+
+                # USE_MY_METRICS
+                if component_lib.get(TRUETYPE_METRICS_KEY, False):
+                    if use_my_metrics_comp is None:
+                        c.flags |= USE_MY_METRICS
+                        use_my_metrics_comp = ufo_component_id
+                    else:
+                        logger.warning(
+                            f"Ignoring USE_MY_METRICS flag on component {i}, "
+                            f"'{ufo_component_id}' because it has been set on "
+                            f"component '{use_my_metrics_comp}' already "
+                            f"in glyph {glyph.name}."
+                        )
+                        c.flags &= ~USE_MY_METRICS
+                else:
+                    c.flags &= ~USE_MY_METRICS
+                lib_contains_use_my_metrics_key |= TRUETYPE_METRICS_KEY in component_lib
```

Comment:
```suggestion
                if TRUETYPE_METRICS_KEY in component_lib:
                    lib_contains_use_my_metrics_key = True
```

I realize this is equivalent, but I had to stop and think when I saw this line 😬